### PR TITLE
Fix newlines in stream title causing text in header to go out of bounds

### PR DESCRIPTION
**Pull request by jupjohn**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

When a stream title has newlines at the end of the text it causes the text in the split header to move upwards. This change just removes the newlines from the end of the string (including whitespace) to prevent this from happening.

### Before:

Chatterino2_Jammehcow_NewlineTitlePre

### After:

Chatterino2_Jammehcow_NewlineTitlePost

### CODE REVIEW

**Comment by jammehcow** [comment]
could almost just replace any newline. would like to test with newlines in the middle of title

**Comment by apa420** [comment]
You can have newlines anywhere in the title so we need to regex all newlines and replace them as spaces like twitch does. Then probably have to run the qtstring function which removes double spaces or w/e.

File: src/widgets/splits/SplitHeader.cpp [hunk 1]
```diff
@@ -170,6 +170,9 @@ namespace {
         if (settings.headerStreamTitle && !s.title.isEmpty())
             title += " - " + s.title;
 
+        // Strip out phantom newlines/trailing whitespace to prevent formatting issues
+        title.replace(QRegExp("[\n\\s]+$"), "");
+
```
Comment (pajlada) [inline src/widgets/splits/SplitHeader.cpp:175]: The 'phantom newlines' can be added through the Twitch website when changing your title by simply pressing enter in the title box.

On the Twitch website, all newlines in the title are stripped.

There's a handy dandy QString function in Qt5 for solving this called `simplified` (see https://doc.qt.io/qt-5/qstring.html#simplified)

```suggestion
        if (settings.headerStreamTitle && !s.title.isEmpty())
        {
            title += " - " + s.title.simplified();
        }

```

**Review by pajlada — CHANGES_REQUESTED** [review]
Small change to support titles like:
```
foo
bar
baz
```

which on Twitch shows up as `foo bar baz`

**Review by pajlada — APPROVED** [review]

**Pull request merged**